### PR TITLE
CPR-131 Nomis additional fields.

### PR DESCRIPTION
**Pull request by vb-moj**
Added end point to get prisoner addresses.

Refactored prisoner service to use prisoner details.

Updated IT.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt [hunk 1]
```diff
@@ -54,7 +59,36 @@ class PrisonerService(
   private fun exactMatchFound(prisonerMatcher: PrisonerMatcher, personEntity: PersonEntity, person: Person) {
     val matchingPrisoner = prisonerMatcher.getMatchingItem()
     logAndTrackEvent(EXACT_MATCH_MESSAGE, TelemetryEventType.NOMIS_MATCH_FOUND, personEntity, person)
-    personRecordService.addPrisonerToPerson(personEntity, matchingPrisoner)
+    val prisonerDetails = getPrisonerDetails(matchingPrisoner.prisonerNumber)
+    if (prisonerDetails != null) {
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt:63]: this bit might as well be in getPrisonDetails. would save an `if` statement

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt [hunk 2]
```diff
@@ -419,4 +419,53 @@ class CourtCaseEventsListenerIntTest : IntegrationTestBase() {
     assertThat(personEntity.offenders[0].aliases[0].surname).isEqualTo("alisSurName")
     assertThat(personEntity.offenders[0].aliases[0].aliasOffenderId).isEqualTo("12345")
   }
+
+  @Test
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt:423]: ❤️

**Review by markreesmoj — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt (hunk 1 above)
Comment (vb-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt:63]: Will update in the next PR.

**Pull request merged**